src/dll/Systems/ScriptCompilationSystem.cpp

```diff
@@ -0,0 +1,110 @@
+#include "ScriptCompilationSystem.hpp"
+#include "stdafx.hpp"
+
+ScriptCompilationSystem::ScriptCompilationSystem(const Paths& aPaths)
+    : m_paths(aPaths)
+    , m_hasScriptsBlob(false)
+{
+}
+
+ESystemType ScriptCompilationSystem::GetType()
+{
+    return ESystemType::Script;
+}
+
+void ScriptCompilationSystem::Startup()
+{
+}
+
+void ScriptCompilationSystem::Shutdown()
+{
+}
+
+void ScriptCompilationSystem::SetScriptsBlob(const std::filesystem::path& aPath)
+{
+    m_scriptsBlobPath = aPath;
+    m_hasScriptsBlob = true;
+}
+
+const std::filesystem::path& ScriptCompilationSystem::GetScriptsBlob() const
+{
+    return m_scriptsBlobPath;
+}
+
+bool ScriptCompilationSystem::HasScriptsBlob() const
+{
+    return m_hasScriptsBlob;
+}
+
+bool ScriptCompilationSystem::Add(std::shared_ptr<PluginBase> aPlugin, const wchar_t* aPath)
+{
+    spdlog::trace(L"Adding path to script compilation: '{}'", aPath);
+    auto resolvedPath = std::filesystem::path(aPath);
+    if (resolvedPath.is_absolute())
+    {
+        if (std::filesystem::exists(resolvedPath))
+        {
+            spdlog::trace(L"Found absolute path: {}", resolvedPath.wstring().c_str());
+            return Add(aPlugin, &resolvedPath);
+        }
+        else
+        {
+            spdlog::error(L"Could not find absolute path: {}", resolvedPath.wstring().c_str());
+            return false;
+        }
+    }
+    else
+    {
+        resolvedPath = aPlugin->GetPath().parent_path() / aPath;
+        if (std::filesystem::exists(resolvedPath))
+        {
+            spdlog::trace(L"Found path relative to plugin: {}", resolvedPath.wstring().c_str());
+            return Add(aPlugin, &resolvedPath);
+        }
+        else
+        {
+            spdlog::error(L"Could not find path relative to plugin: {}", resolvedPath.wstring().c_str());
+            return false;
+        }
+    }
+}
+
+bool ScriptCompilationSystem::Add(std::shared_ptr<PluginBase> aPlugin, std::filesystem::path* aPath)
+{
+    std::scoped_lock _(m_mutex);
+    m_scriptPaths.emplace(aPlugin, std::move(*aPath));
+    return true;
+}
+
+FixedWString ScriptCompilationSystem::GetCompilationArgs(const FixedWString& original)
+{
+    auto buffer = fmt::wmemory_buffer();
+    if (this->m_hasScriptsBlob)
+    {
+        spdlog::info("Using scriptsBlobPath");
+        format_to(std::back_inserter(buffer), LR"(-compile "{}" "{}")", m_paths.GetR6Scripts().wstring(),
+                  m_scriptsBlobPath.wstring());
```

Comment:
If you include `Utils.hpp` then you can do this:
```suggestion
        format_to(std::back_inserter(buffer), LR"(-compile "{}" "{}")", m_paths.GetR6Scripts(), m_scriptsBlobPath);
```